libsql_client/hrana/convert.py

```diff
@@ -105,10 +105,12 @@ def _value_to_proto(in_value: InValue) -> proto.Value:
         if not math.isfinite(value):
             raise ValueError("Only finite floats (not Infinity or NaN) are supported")
         return {"type": "float", "value": value}
-    elif isinstance(value, bytes):
-        return {"type": "blob", "base64": base64.b64encode(value).decode()}
     else:
-        raise TypeError(f"Unsupported value of type {type(value)}")
+        try:
+            data = base64.b64encode(value).decode()
+            return {"type": "blob", "base64": data }
+        except TypeError:
+            raise TypeError(f"Unsupported value of type {type(value)}")
```

Comment:
The logic that converts various byte objects into `bytes` should be in `_normalize_value()`. If the user passes an unsupported object (such as a dict or instance of their class), it will be extremely suprising if we try to convert it to base64, and it may produce weird results.

Reply:
Or, is there a way to reliably recognize objects that implement the buffer interface? That might be the best solution.

Reply:
don't worry, it will go `TypeError`:

```py
>>> import base64
>>> base64.b64encode({})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/base64.py", line 58, in b64encode
    encoded = binascii.b2a_base64(s, newline=False)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: a bytes-like object is required, not 'dict'
```